database/layer.go

```diff
@@ -131,6 +131,59 @@ func InsertLayer(layer *Layer) error {
 	return nil
 }
 
+// DeleteLayer delete the specified layer and every layers that are based on that one, recursively.
```

Comment:
DeleteLayer deletes the specified layer and any child layers that are dependent on the specified layer.